include/mbedtls/config.h

```diff
@@ -1026,6 +1026,16 @@
  */
 //#define MBEDTLS_SHA512_SMALLER
 
+/**
+ * \def MBEDTLS_SHA512_NO_SHA384
+ *
+ * Disable the SHA-384 option of the SHA-512 module. Use this to save some
+ * code size on devices that don't use SHA-384.
+ *
+ * Uncomment to disable SHA-384
+ */
+//#define MBEDTLS_SHA512_NO_SHA384
+
```

Comment:
Regarding `NO` vs `WITH`, I agree in principle, but unfortunately there's a backwards-compatibility issue. Currently people who use a custom `config.h` can just define `MBEDTLS_SHA512_C` and they get SHA-384 as well. If make SHA-384 available only if `MBEDTLS_SHA512_WITH_SHA384` is defined as well, it means they have to change their code (namely, their `config.h`) when updating Mbed TLS, which one thing we promised they won't have to do except on major version upgrades.

So in Mbed TLS 3.0.0 I agree with should have a `WITH` option rather than a `NO` option. But in the meantime if we want to merge for a 2.x release, I'm afraid a `NO` option is our only choice.

What we could to however, is to translate it to a `WITH` option internally, so that we at least avoid the readability issue inside the `.c` file.

Reply:
It makes sense to me. Thanks for explanation